src/publisher.impl.js

```diff
@@ -32,17 +32,35 @@ var execWithInheritedStdio = function (command, callback) {
 var Publisher = function (options) {
     options = options || {};
     this.compress = options.compress === undefined ? true : options.compress;
+    this.noClobber = !! options.noClobber;
 };
 
 Publisher.prototype.publish = function (srcPath, dstPath, doneCallback) {
     var fns = [];
 
+    if (this.noClobber) {
+        fns.push(function (callback) {
+            var command = 's3 ls ' + dstPath;
```

Comment:
Not sure this is accomplishing quite what it's intended. If you sync a folder and pass `--noClobber`, what I'd expect is that no files would be deleted. In this case, it seems like you will only guarantee that if what you're syncing is a single key.